go-controller/pkg/cluster/node.go

```diff
@@ -74,59 +74,65 @@ func (cluster *OvnClusterController) StartClusterNode(name string) error {
 	var subnet *net.IPNet
 	var clusterSubnets []string
 	var cidr string
-	var connected bool
+	var readyChan = make(chan bool, 1)
+
+	err = cluster.watchConfigEndpoints(readyChan)
+	if err != nil {
+		return err
+	}
+
+	// Hold until we are certain that the endpoint has been setup.
+	// We risk polling an inactive master if we don't wait while a new leader election is on-going
+	<-readyChan
```

Comment:
@alexanderConstantinescu if `config.OvnNorth.Address` and `config.OvnSouth.Address` are already set, then we don't need to block waiting for the `readyChan` because the user explicitly set the DBs on the command/line config and we let them shoot themselves in the foot.

But if the address is empty, then yes we should wait.

Reply:
> But if the address is empty, then yes we should wait.

That's what I had seen. And I assume it's because - and correct me if I am wrong - the watcher just above will ultimately be spun off in a go-routine, we thus risk starting to poll the node for its logical switch just below with an empty ovnkube master IP, and thus failing. We thus need to wait until leader election is done and we are notified of the proper endpoint, as to communicate correctly with the right master.